test/e2e/apps/daemon_set.go

```diff
@@ -329,6 +329,10 @@ var _ = SIGDescribe("Daemon set [Serial]", func() {
 		framework.Logf("Creating simple daemon set %s", dsName)
 		ds := newDaemonSet(dsName, image, label)
 		ds.Spec.UpdateStrategy = apps.DaemonSetUpdateStrategy{Type: apps.RollingUpdateDaemonSetStrategyType}
+		// Set the termination grace period to 1 second. This is necessary because the
+		// serve-hostname binary sleeps for 1 minute after receiving SIGTERM.
+		terminationPeriod := int64(1)
+		ds.Spec.Template.Spec.TerminationGracePeriodSeconds = &terminationPeriod
```

Comment:
This fix should work, but DaemonSet test shouldn't be tied to the serve-hostname image. We can simply change the image we use in this test:
```
image := framework.ServeHostnameImage
```
to something else (can pick from [this list](https://github.com/kubernetes/kubernetes/blob/a2a5bd03fdae6ac45509560157f4c24d1c001e86/test/utils/image/manifest.go#L93)), such as nginx.

We can also increase the timeout based on number of nodes, as this affects the number of replicas a DaemonSet creates.

Reply:
That is a good point. Should we take that route instead? Will take a look at the list of images.

Reply:
Yes, thanks for looking into this!

Reply:
Should we update the image used throughout all DaemonSet tests, or just the one used during the DaemonSet update test?

Reply:
Updating the image used throughout all DaemonSet tests (i.e. change `image := framework.ServeHostnameImage`) is fine. Which image we pick doesn't matter. 

We can scale update timeout based on # of nodes too, maybe something like `5 mins + 1 min * # of nodes`.